fastavro/writer.py

```diff
@@ -425,7 +425,11 @@ def dump():
     acquaint_schema(schema)
 
     for record in records:
-        write_data(io, record, schema)
+        try:
```

Comment:
I need to think about this one. Looks like a good start, however:
* The name `validate_data` does not really fit what the function is doing.
* The error raised by `validate_data` might mask the original error (`Exception`)
* `try/catch` has performance penalty, we need to benchmark